### PR TITLE
Headbite Changes

### PR DESCRIPTION
**Pull request by Hathkar**
## About The Pull Request

- [x] Can't be used while on fire
- [x] Channel will cancel if damage is taken
- [x] Only usable on the dead

## Why It's Good For The Game

Lots of people want changes, so here they are.

## Changelog
:cl:
balance: Headbite is only usable on the dead
balance: Headbite can't be used while on fire
balance: Headbite will cancel if damage is taken
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
did ya find a way to cancel it if takes damage ?

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -62,6 +62,13 @@
 		return FALSE
 	if(!X.Adjacent(victim)) //checks if owner next to target
 		return FALSE
+	if(X.on_fire)
+		to_chat(X, "<span class='warning'>We're too busy being on fire to do this!</span>")
+		return FALSE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:67]: silent check

**Comment by Wayland-Smithy** [comment]
I suppose adding the requirement they be dead is fine, but headbiting for the kill is great fun, but...
> Channel will cancel if damage is taken

Is there any precedent for this? we already have stagger, do marines who are building/busy doing other stuff get their action canceled on damage cause of any "ouchies the pain causes you to mess up what you are doing" type reasoning?

For non high-speed castes the risk/reward of going for a headbite with marines close by is already good gameplay IMO. Maybe headbiting can "exaust" the xeno for a couple seconds after channel end by capping move speed to that of like queen or crusher? It could have a ui element like purp gas or broken bones.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 1 above)
Comment (Hughgent) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:67]: i.e. put this behind `if(!silent)`

**Review by Hughgent — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**